storage/database/src/backend/heed/env.rs

```diff
@@ -282,10 +282,32 @@ where
         tx_rw: &RefCell<heed::RwTxn<'env>>,
     ) -> Result<impl DatabaseRw<T>, RuntimeError> {
         // Open up a read/write database using our table's const metadata.
-        Ok(HeedTableRw {
-            db: self.create_database(&mut tx_rw.borrow_mut(), Some(T::NAME))?,
-            tx_rw,
-        })
+        let db = {
+            let mut tx_rw = tx_rw.borrow_mut();
+            let mut db = self.database_options();
+
+            // Set the key comparison behavior.
+            match <T::Key>::KEY_COMPARE {
+                // Already the default for LMDB, setting
+                // a custom comparison function is slower.
+                KeyCompare::Lexicographic => (),
+
+                // Instead of setting a custom [`heed::Comparator`],
+                // use this LMDB flag; it is ~10% faster.
+                KeyCompare::Number => {
+                    db.flags(DatabaseFlags::INTEGER_KEY);
+                }
+
+                // Use a custom comparison function if specified.
+                KeyCompare::Custom(_) => {
+                    db.key_comparator::<T::Key>();
+                }
+            }
+
+            db.types().name(T::NAME).create(&mut tx_rw)?
+        };
```

Comment:
## Key 4/4
...we forward the `KeyCompare::Number` to use the faster flag instead. We only fall back to custom integer comparison if the number isn't supported.

Our only number keys for `cuprate_blockchain` are `u64` so it doesn't really matter to support this for _us_, but I thought it would be nice to maintain the sorted behavior for _all_ numbers, like `redb`.